HandleNHCreateOrUpdateInstallationCallOrchestrator/handler.ts

```diff
@@ -70,14 +72,58 @@ export const getHandler = ({
 
       if (isUserATestUser.value) {
         logger.info(`TEST_USER:${installationId}`);
-      }
 
-      yield* createOrUpdateActivity(context, {
-        installationId,
-        notificationHubConfig: legacyNotificationHubConfig,
-        platform,
-        pushChannel,
-        tags
-      });
+        const notificationHubConfigPartition = notificationHubConfigPartitionChooser(
+          installationId
+        );
+
+        try {
+          yield* createOrUpdateActivity(context, {
+            installationId,
+            notificationHubConfig: notificationHubConfigPartition,
+            platform,
+            pushChannel,
+            tags
+          });
+
+          // Always delete installation from legacy Notification Hub
+          yield* deleteInstallationActivity(context, {
+            installationId,
+            notificationHubConfig: legacyNotificationHubConfig
+          });
+        } catch (err) {
+          // ^In case of exception, delete from partition and restore into legacy NH
+
+          logger.error(
+            failureUnhandled(
+              `ERROR|TEST_USER ${installationId}: ${toString(err)}`
+            )
+          );
+
+          yield* createOrUpdateActivity(context, {
+            installationId,
+            notificationHubConfig: legacyNotificationHubConfig,
+            platform,
+            pushChannel,
+            tags
+          });
+
+          yield* deleteInstallationActivity(context, {
+            installationId,
+            notificationHubConfig: notificationHubConfigPartition
+          });
```

Comment:
Is compensating activites really required here?
The upsert on new NH# could:
- success -> ok
-  fail -> nothing added (nothing to compense for).
If the delete on old NH0 fails, in the wrost scenario the notification will be pushed 2 times.

There is no feasible way to retrieve the "migration status" from Azure Notification Hub, so the compensation will be applied also for already migrated one, reverting it back on the old NH0.

Reply:
@balanza Fabrizio and me talked about this yersterday, and we agreed about what he wrote here. What do you think?
Just as a reminder, we never had an Orchestrator failed because of a createOrUpdate activity failure till now.

Reply:
If I get your arguments right, I'd say the issue is we are considering `delete(NH0)` to be a blocking failure which must be handled. I disagree on that. Fallback is needed only when `create(NH-X)` fails, as we want its installation to be stored in `NH0` as a remediation.

|_when this fails_| _then we do_|
|-|-|
|`deleteInstallation(NH0)`|ignore the error, just trace it for stats|
|`createOrUpdate(NH-X)`|do `createOrUpdateActivity(NH0)` so the Citizen will still receive notifications (notify activity acts in parallel on both `NH-X` and `NH-0`)|
|`createOrUpdate(NH0)`|it's actually a critical error as it means we failed in the remediation as well, thus we must consider the whole operation failed|

Reply:
@balanza @fabriziopapi 
Not sure about the remediation in case of `createOrUpdate(NH-X)` failure. 
Let me explain: the only moment in which we can actually loose the Citizen's Installation is during the first migration. After that, we will have his registration safely stored into NH-X even if we fail a single `createOrUpdate`.
If we decide to make that kind of remediation, we are actually bringing back his installation into NH0. And due to the huge number of createOrUpdate we have, this scenario has a higher probability. 
Just as a reminder: we've never had an Orchestrator failure because of a `createOrUpdate` Activity failure.

Reply:
About the createOrUpdate(NH-X), we explored 2 scenario:
1 - new user
2 - existing user

1 - New user must not be create on old NH0, so no remediation needed (if a new installation do not success on NH-X, it will not success on NH-0, too)
2 - Existing user already existing on NH0. If createOrUpdate(NH-X) fails, the deleteInstallation(NH0) will not be called, so no remediation needed.

Reply:
> New user must not be create on old NH0

They are now. The point is to have a lossless transition, if we reckon we're doing something wrong, it's not a problema to fallback.

> user already existing on NH0

Not actually, `device_token` may have changed and we could have an expired value.

---

> we are actually bringing back his installation into NH0

I see the point and I agree. Let's just consider that the fallback mechanism (and the consequent remediation) is intended for the rollout phase only, during which we have to monitor if changes we introduced are safe ([step 3](https://excalidraw.com/#json=6302097846829056,V_vrLMGC-Nql0nxQ9nqrCA)). Once we're done, and we are confident the partitioning logic introduced no misbehaviours, we can get rid of it along with the feature-flag condition ([step 4](https://excalidraw.com/#json=4910240143245312,heN_LutW1XHyX7gGuQzP2Q)).

Reply:
Fine for me